### ISSUE TITLE
Std::time::Instant should not be Send?

### ISSUE THREAD

**Issue by omerbenamram** [rust-lang/rust#85928]
Hi,

I'd like to preface that we're not entirely sure this is a "bug" - but rather a surprising behavior of rust.
We've been experiencing some panics in production with code that kept some `std::time::Instant` in a struct and updated it across threads (using `Arc<RwLock>>`).

minimal reproducible code:

```rust
use std::thread;
use std::{
    sync::{Arc, RwLock},
    time::Instant,
};
fn main() {
    let shared_key = Arc::new(RwLock::new(Instant::now()));
    let writer_key = shared_key.clone();
    thread::spawn(move || loop {
        *writer_key.write().unwrap() = Instant::now();
    });
    loop {
        let _ = Instant::now() - *shared_key.read().unwrap();
    }
}
```

Will panic almost immediately (on linux):
```
thread 'main' panicked at 'supplied instant is later than self', library/std/src/time.rs:281:48
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The fix we've applied was using `saturating_duration` for now - but this raises a question regarding the underlying assumption was that the clock will retain monotonicity across threads, this is not the case however.
[Rust docs](https://doc.rust-lang.org/std/time/struct.Instant.html) do not mention the threading issue directly.

[This SO post ](https://stackoverflow.com/questions/64159779/rust-stdtimeinstant-panicked-at-supplied-instant-is-later-than-self) raised our suspicion regarding the multi-core issue.

The [documentation for linux](https://linux.die.net/man/3/clock_gettime) mentions:

> The CLOCK_PROCESS_CPUTIME_ID and CLOCK_THREAD_CPUTIME_ID clocks are realized on many platforms using timers from the CPUs (TSC on i386, AR.ITC on Itanium). These registers may differ between CPUs and as a consequence these clocks may return bogus results if a process is migrated to another CPU.
> 
> If the CPUs in an SMP system have different clock sources then there is no way to maintain a correlation between the timer registers since each CPU will run at a slightly different frequency. If that is the case then clock_getcpuclockid(0) will return ENOENT to signify this condition. The two clocks will then only be useful if it can be ensured that a process stays on a certain CPU.
> 

In this case - should `std::time::Instant` considered `Send`?  Since the underlying guarantee of monotonicity is not upheld across threads. Or should this just be a docs mention?

Thanks!

**Comment by omerbenamram** [comment]
Update: This was a race condition in `Instant::now() - *shared_key.read().unwrap()`. Closing this :)
